Update feature_space.py

orbin123 opened the pull request:
Allow `adapt()` to accept Python iterables in FeatureSpace.

* FeatureSpace: Updated `adapt()` to detect dictionary inputs (of lists/arrays) and automatically convert them to tf.data.Dataset. Previously, it yielded a ValueError demanding a Dataset object.

This change enable users to pass standard Python data structures directly to the preprocessing layer, improving usability and consistency with other layers like `IntegerLookup`.

orbin123 commented:
Could gemini explain the issue. Is the issue saying there should be a human reviewer to push or is there any other configurations

hertschuh reviewed (commented):
Hi @orbin123 

Thanks for the PR.

Can you add unit tests covering the changes in `feature_space.py`?

orbin123 commented:
Thankyou for the opportunity
I have reverted the changes in normalisation file as you have suggested and added the unit test function for the update.

hertschuh reviewed (approved):
Thanks for the update.
